packages/vscode-dependencies-validation/src/autofix/depsIssues.ts

```diff
@@ -0,0 +1,59 @@
+import type { Uri } from "vscode";
+import {
+  VscodeFileEventConfig,
+  VscodeUriFile,
+  VscodeWorkspace,
+} from "../vscodeTypes";
+import {
+  ENABLE_AUTOFIX,
+  getAutoFixDelay,
+  isAutoFixEnabled,
+} from "./configuration";
+import { addPackageJsonFileWatcher as addPackageJsonFileWatcher } from "./packageJsonFileWatcher";
+import { addUnsupportedFilesWatcher } from "./unsupportedFilesWatcher";
+import { findAndFixDepsIssues } from "../util";
+import { clearDiagnostics } from "../util";
+
+export function activateDepsIssuesAutoFix(
+  vscodeConfig: VscodeFileEventConfig & VscodeUriFile
+): void {
+  fixWorkspaceDepsIssues(vscodeConfig);
+  addPackageJsonFileWatcher(vscodeConfig);
+  addUnsupportedFilesWatcher(vscodeConfig);
+
+  onAutoFixChange(vscodeConfig);
+}
+
+function fixWorkspaceDepsIssues(vscodeConfig: VscodeFileEventConfig): void {
+  setTimeout(() => {
+    void findDepsIssuesFixThemAndCleanProblemsView(vscodeConfig);
+  }, getAutoFixDelay(vscodeConfig.workspace));
+}
+
+function getPackageJsonUris(workspace: VscodeWorkspace): Thenable<Uri[]> {
+  return workspace.findFiles("package.json", "**​/node_modules/**");
+}
+
+function onAutoFixChange(vscodeConfig: VscodeFileEventConfig): void {
+  vscodeConfig.workspace.onDidChangeConfiguration((event) => {
+    const affected = event.affectsConfiguration(ENABLE_AUTOFIX);
```

Comment:
logically a configuration change may only require handling if the value was changed to
`true`? 

I think the above statement may be more correct if we remove any problems view handling from this flow.

Reply:
it will work only if auto fix is true, see findDepsIssuesFixThemAndCleanProblemsView method